node_package/components/asset_field_type.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+const ToolTip = ({id, tooltip}) => (<p>{ tooltip }</p>)
+
+const allowedExtensions = ({allowed_extensions}) => allowed_extensions === undefined ? '' : allowed_extensions.join(', ')
+const allowedExtensionsForFor = ({allowed_extensions}) => allowed_extensions === undefined ? '' : '.' + allowed_extensions.join(',.')
+const maxFileSize = ({max_size}) => max_size
+
+class AssetFieldType extends React.PureComponent {
+  render() {
+    const { field_item, id, metadata, required=false, name, validations } = this.props
+    console.log('AssetFieldType', this.props)
+    return (
+      <div>
+        <strong>Allowed extensions: </strong>
+          { allowedExtensions(validations) }
+        <br/>
+        <strong>Allowed filesize: </strong>
+        { maxFileSize(validations) }
+        <br />
+        <input type='hidden' value={id} />
+        { field_item.tooltip && ToolTip(field_item) }
+        <label>{ name } </label>
```

Comment:
add a `for`/`htmlFor` attribute to the `<label>` with a value of `name`

Or alternatively, wrap the `label` around the `input` and close `</label>` after the `<input>` like

```
<label><input></label>
```

Either of these will tie the label and the input together.  Do the same for the other inputs, like in `date_time_type.jsx`

Reply:
Oh yeah, will do. We will probably be pulling in some type of form library in the future.